### PR TITLE
FEC-11372 HLS Widevine Offline Support

### PR DESCRIPTION
**Pull request by GouravSna**

### CODE REVIEW

File: dtglib/src/main/java/com/kaltura/dtg/hls/HlsDownloader.java [hunk 1]
```diff
@@ -89,6 +99,7 @@ public void createDownloadTasks() throws IOException {
                 order++;
                 maybeAddTask(tasks, track.getRelativeId(), trackTargetDir, chunk.lineNum, chunk.ext, chunk.url, order);
                 maybeAddTask(tasks, track.getRelativeId(), trackTargetDir, chunk.encryptionKeyLineNum, chunk.keyExt, chunk.encryptionKeyUri, order);
+                mayBeAddInitSegment(tasks, track, trackTargetDir, order);
```
Comment (GouravSna) [inline dtglib/src/main/java/com/kaltura/dtg/hls/HlsDownloader.java]: Call it 'maybeAddInitSegmentTask'

File: dtglib/src/main/java/com/kaltura/dtg/hls/HlsDownloader.java [hunk 2]
```diff
@@ -58,6 +58,16 @@ private static void maybeAddTask(LinkedHashSet<DownloadTask> tasks, String relat
         tasks.add(task);
     }
 
+    private static void mayBeAddInitSegment(LinkedHashSet<DownloadTask> tasks, HlsAsset.Track track, File trackTargetDir, int order) {
+        if (track.initSegment != null && track.initSegment.size() > 0) {
+            while (track.initSegment.keySet().iterator().hasNext()) {
+                HlsAsset.Chunk initChunk = track.initSegment.values().iterator().next();
+                maybeAddTask(tasks, track.getRelativeId(), trackTargetDir, initChunk.lineNum, initChunk.ext, initChunk.url, order);
+                track.initSegment.clear();
```
Comment (giladna) [inline dtglib/src/main/java/com/kaltura/dtg/hls/HlsDownloader.java]: @GouravSna  please change the code so we do not touch the data structure during iteration

File: dtglib/src/main/java/com/kaltura/dtg/hls/HlsDownloader.java (hunk 1 above)
Comment (GouravSna) [reply dtglib/src/main/java/com/kaltura/dtg/hls/HlsDownloader.java]: Done

File: dtglib/src/main/java/com/kaltura/dtg/hls/HlsDownloader.java (hunk 2 above)
Comment (GouravSna) [reply dtglib/src/main/java/com/kaltura/dtg/hls/HlsDownloader.java]: Done. No need to clear the map itself.

Comment (GouravSna) [reply dtglib/src/main/java/com/kaltura/dtg/hls/HlsDownloader.java]: Changed the loop with clearing the map.

**Review by giladna — APPROVED** [review]

**Pull request merged**